src/cli-validator/utils/processConfiguration.js

```diff
@@ -367,9 +367,39 @@ const validateConfigOption = function(userOption, defaultOption) {
   return result;
 };
 
+const getSpectralRuleset = async function(defaultRuleset) {
+  // List of ruleset files to search for
+  const ruleSetFilesToFind = [
+    '.spectral.yaml',
+    '.spectral.yml',
+    '.spectral.json'
+  ];
+  let ruleSetFile;
+
+  // search up the file system for the first ruleset file found
+  try {
+    for (const file of ruleSetFilesToFind) {
+      if (!ruleSetFile) {
+        ruleSetFile = await findUp(file);
+      }
```

Comment:
It looks like the files will be searched in prioritized order. I think this is a good approach, but maybe it should be documented

Reply:
Come to think of it, it doesn't look like there are any README changes. Is there anything the user would need to know to take advantage of this new feature? We should probably add a section about it - even if it just tells the user what's available and links the appropriate Spectral documentation

Reply:
I'm fine with adding some documentation on at least how to specify your own custom ruleset file. Though I had the idea that this initial implementation would be "hidden" from the user.

Reply:
I guess this was intended to be a "Beta" feature for the time being, so that makes sense to me. Disregard this comment